### ISSUE TITLE
Analyzer Feedback from IntelliJ

### ISSUE THREAD

**Issue by RobinJespersen** [dart-lang/sdk#37996]
Analyzer Feedback from IntelliJ

## Version information

- `IDEA AI-191.8026.42.35.5791312`
- `2.5.0-edge.2023f09b561c59920301dd27ee22fd44860c9562`
- `AI-191.8026.42.35.5791312, JRE 1.8.0_202-release-1483-b49-5587405x64 JetBrains s.r.o, OS Mac OS X(x86_64) v10.14.6, screens 1920x1200  Retina`

## Exception
```
Dart analysis server, SDK version 2.5.0-edge.2023f09b561c59920301dd27ee22fd44860c9562, server version 1.27.2, error: Exception while using a PreferConstConstructors to visit a InstanceCreationExpressionImpl in ArgumentListImpl in MethodInvocationImpl in AwaitExpressionImpl in ExpressionStatementImpl in BlockImpl in BlockFunctionBodyImpl in FunctionExpressionImpl in ArgumentListImpl in MethodInvocationImpl in ExpressionStatementImpl in BlockImpl in BlockFunctionBodyImpl in FunctionExpressionImpl in FunctionDeclarationImpl in CompilationUnitImpl
NoSuchMethodError: The getter 'type' was called on null.
Receiver: null
Tried calling: type
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      VariableMember.type (package:analyzer/src/dart/element/member.dart:1053:61)
#2      ConstantEvaluationEngine.evaluateConstructorCall (package:analyzer/src/dart/constant/evaluation.dart:626:40)
#3      ConstantVerifier.visitInstanceCreationExpression (package:analyzer/src/dart/constant/constant_verifier.dart:134:27)
#4      InstanceCreationExpressionImpl.accept (package:analyzer/src/dart/ast/ast.dart:6180:15)
#5      LinterContextImpl._hasConstantVerifierError (package:analyzer/src/lint/linter.dart:322:10)
#6      LinterContextImpl.canBeConst (package:analyzer/src/lint/linter.dart:290:15)
#7      _Visitor.visitInstanceCreationExpression (package:linter/src/rules/prefer_const_constructors.dart:95:19)
#8      InstanceCreationExpressionImpl.accept (package:analyzer/src/dart/ast/ast.dart:6180:15)
#9      LinterVisitor._runSubscriptions (package:analyzer/src/lint/linter_visitor.dart:713:14)
#10     LinterVisitor.visitInstanceCreationExpression (package:analyzer/src/lint/linter_visitor.dart:383:5)
#11     InstanceCreationExpressionImpl.accept (package:analyzer/src/dart/ast/ast.dart:6180:15)
#12     NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:7670:20)
#13     ArgumentListImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:377:16)
#14     RecursiveAstVisitor.visitArgumentList (package:analyzer/dart/ast/visitor.dart:623:10)
#15     ArgumentListImpl.accept (package:analyzer/src/dart/ast/ast.dart:373:49)
#16     MethodInvocationImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:7240:20)
#17     RecursiveAstVisitor.visitMethodInvocation (package:analyzer/dart/ast/visitor.dart:1067:10)
#18     LinterVisitor.visitMethodInvocation (package:analyzer/src/lint/linter_visitor.dart:456:11)
#19     MethodInvocationImpl.accept (package:analyzer/src/dart/ast/ast.dart:7233:49)
#20     AwaitExpressionImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:920:18)
#21     RecursiveAstVisitor.visitAwaitExpression (package:analyzer/dart/ast/visitor.dart:653:10)
#22     LinterVisitor.visitAwaitExpression (package:analyzer/src/lint/linter_visitor.dart:54:11)
#23     AwaitExpressionImpl.accept (package:analyzer/src/dart/ast/ast.dart:916:49)
#24     ExpressionStatementImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:3770:18)
#25     RecursiveAstVisitor.visitExpressionStatement (package:analyzer/dart/ast/visitor.dart:833:10)
#26     Lin

**Comment by MichaelRFairhurst** [comment]
Just caught this with the fuzzer, giving us a repro: #38204

**Comment by scheglov** [comment]
Fixed by c6b0ae1fdab360f08b7966797506dae6629551fd.
